.github/workflows/self_check.yml

```diff
@@ -0,0 +1,22 @@
+---
+name: Self Checks
+on: pull_request
+
+jobs:
+  actionlint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```

Comment:
It appears dependabot (as well as Renovate) supports doing this for various formats...
https://github.blog/changelog/2022-10-31-dependabot-now-updates-comments-in-github-actions-workflows-referencing-action-versions/

I think given the use of this repo, it's probably best to pin all the workflows? I can make a separate PR to do that for the reusable ones if this format is preferred and if someone would like.

Reply:
I'm not a fan of the exact pinning because you get so many changes just to update some version. Reading through the history gets quite hard IMHO.

Reply:
btw, I took them out here. But, if it's decided at some point that it's wise to pin them, let me know and I can throw up a PR doing that.

Reply:
Let me know if that looks better now.